pkg/store/store_test.go

```diff
@@ -178,6 +179,14 @@ func TestStoreAPI(t *testing.T) {
 		}
 	})
 
+	// Test Delete Non Existent Model
+	t.Run("Delete", func(t *testing.T) {
```

Comment:
Maybe worth checking in `client_test.go` instead, to ensure the type can be checked as a `distribution.ErrModelNotFound`?

```
func TestClientDeleteNonexistentModel(t *testing.T) {
	// Create temp directory for store
	tempDir, err := os.MkdirTemp("", "model-distribution-test-*")
	if err != nil {
		t.Fatalf("Failed to create temp directory: %v", err)
	}
	defer os.RemoveAll(tempDir)

	// Create client
	client, err := NewClient(WithStoreRootPath(tempDir))
	if err != nil {
		t.Fatalf("Failed to create client: %v", err)
	}

	// Delete the model
	if err := client.DeleteModel("some/missing:model"); !errors.Is(err, ErrModelNotFound) {
		t.Fatalf("Should return ErrModelNotFound got: %v", err)
	}
}
```
We should really make `store` and most other packages internal